src/chunk.c

```diff
@@ -2194,34 +2203,120 @@ ts_chunk_show_chunks(PG_FUNCTION_ARGS)
 		else
 			time_type = InvalidOid;
 
+		/* note that arg_types will be the same for all specified "ANY" elements for a given call */
+		arg_type = InvalidOid;
 		if (!PG_ARGISNULL(1))
-			older_than = ts_time_value_from_arg(PG_GETARG_DATUM(1),
-												get_fn_expr_argtype(fcinfo->flinfo, 1),
-												time_type);
+		{
+			arg_type = get_fn_expr_argtype(fcinfo->flinfo, 1);
+			older_than = ts_time_value_from_arg(PG_GETARG_DATUM(1), arg_type, time_type, true);
+			older_newer = true;
+		}
 
 		if (!PG_ARGISNULL(2))
-			newer_than = ts_time_value_from_arg(PG_GETARG_DATUM(2),
-												get_fn_expr_argtype(fcinfo->flinfo, 2),
-												time_type);
+		{
+			arg_type = get_fn_expr_argtype(fcinfo->flinfo, 2);
+			newer_than = ts_time_value_from_arg(PG_GETARG_DATUM(2), arg_type, time_type, true);
+			older_newer = true;
+		}
+
+		/*
+		 * We cannot have a mix of [older_than/newer_than] and [created_before/created_after].
+		 * So, check that first.
+		 */
+
+		if (!PG_ARGISNULL(3))
+		{
+			if (older_newer)
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("cannot specify \"older_than\" or \"newer_than\" together with "
+								"\"created_before\""
+								"or \"created_after\"")));
+
+			arg_type = get_fn_expr_argtype(fcinfo->flinfo, 3);
+			created_before = ts_time_value_from_arg(PG_GETARG_DATUM(3), arg_type, time_type, false);
+			before_after = true;
+		}
+
+		if (!PG_ARGISNULL(4))
+		{
+			if (older_newer)
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("cannot specify \"older_than\" or \"newer_than\" together with "
+								"\"created_before\""
+								"or \"created_after\"")));
+
+			arg_type = get_fn_expr_argtype(fcinfo->flinfo, 4);
+			created_after = ts_time_value_from_arg(PG_GETARG_DATUM(4), arg_type, time_type, false);
+			before_after = true;
+		}
+
+		/* if both have not been specified then default to older_newer */
+		if (!older_newer && !before_after)
+			older_newer = true;
 
 		funcctx = SRF_FIRSTCALL_INIT();
-		funcctx->user_fctx = get_chunks_in_time_range(ht,
-													  older_than,
-													  newer_than,
-													  "show_chunks",
+		/*
+		 * For INTEGER type dimensions, we support querying using intervals or any
+		 * timetamp or date input. For such INTEGER dimensions, we get the chunks
+		 * using their creation time values.
+		 */
+		if (IS_INTEGER_TYPE(time_type) && (arg_type == INTERVALOID || IS_TIMESTAMP_TYPE(arg_type)))
+		{
+			/* check that we use proper inputs for such cases */
+			if (older_newer)
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("cannot specify \"older_than\" and/or \"newer_than\" for "
+								"\"integer\"-like partitioning types"),
+						 errhint(
+							 "Use \"created_before\" and/or \"created_after\" which rely on the "
+							 "chunk creation time values.")));
```

Comment:
I think you removed these hints above, but not here and other places (e.g., below). Make sure it is consistent.

Reply:
@erimatnor the first one is a legitimate error because we cannot use `older_than` or `newer_than` for non-time partitioning columns.
Removed the second occurrence which was a WARNING.